### PR TITLE
feat: Add frontend development overview and useful resources

### LINKED ISSUE #53: Improve linked content with descriptions and optionally an image
# Suggestion/Improvement

Currently, our links to external resources for learning are not very descriptive. Improve one or more links by adding a description, and optionally a screenshot or gif that provides more context for readers to the linked content.

## Description

For example, a link that says only "Codum" can be improved to provide more information in the link itself such as:

"Codum engineering accountability buddy platform"
Codum runs multiple cohorts a year. They pair individuals who form a team of two. These two define a goal for themselves and hold each other accountable for achieving that goal within the timeframe of the cohort.

## Motivation

We have great information on other pages of the knowledge-hub, we should raise the quality of our links to external content, to match other information.

## Implementation (if applicable)

Choose one or more links to improve.

### PR DESCRIPTION
**Pull request by RitaDee**
## Assigned Issue

- [ ] No issue assignment necessary.
- [x] I am assigned to the related issue below.

This PR Closes #53 

## Description
This PR adds a description to external learning resource links. I enhanced these links by incorporating detailed descriptions to give users better context.

Also, the addition of the "Graphical Design" and "UI Development" sections in this PR for the "Frontend Development" chapter serves to enrich the content and provide a more comprehensive understanding of frontend development. These sections provide readers with a deeper understanding of the various aspects of frontend development beyond just coding. Graphical design and UI development are integral components of frontend development, and understanding their significance helps developers create visually appealing and user-friendly applications.

## Type of PR

- [ ] ✨ Feature
- [ ] 🐞 Bug fix
- [x] 📄 Documentation update
- [ ] Others

### CODE REVIEW

**Comment by mariahlaqua** [comment]
Thank you @RitaDee !! 👍 💯 

Will review this today!

**Comment by RitaDee** [comment]
> Thank you @RitaDee !! 👍 💯
> 
> Will review this today!

Okay.

File: resources/frontend-development.md [hunk 1]
```diff
@@ -1,14 +1,70 @@
 # Frontend Development
 
+Frontend Development is a pivotal aspect of web development that focuses on crafting the visual and interactive components of a web application. It encompasses two fundamental elements: the graphical design and the user interface (UI) development.
+
+## Graphical Design
+
+In the context of frontend development, graphical design involves conceptualizing and creating the visually appealing layouts, typography,color schemes, and graphical elements that collectively form the aesthetic foundation of the application.Designers employ tools like Adobe Photoshop, Sketch, or Figma to translate ideas into tangible visual assets.
```
Comment (mariahlaqua) [inline resources/frontend-development.md:7]: Just some small type changes for readability here:

`In the context of frontend development, graphical design involves conceptualizing and creating the visually appealing layouts, typography, color schemes, and graphical elements that collectively form the aesthetic foundation of the application. Designers employ tools like Adobe Photoshop, Sketch, or Figma to translate ideas into tangible visual assets.`

**Review by ghost — CHANGES_REQUESTED** [review]
This looks wonderful! I added a comment for some missing spaces in the copy.

Thanks so much @RitaDee !

**Comment by RitaDee** [comment]
> This looks wonderful! I added a comment for some missing spaces in the copy.
> 
> Thanks so much @RitaDee !

Thank you, @mariahlaqua, for the review. I  will make the changes.

File: resources/frontend-development.md (hunk 1 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done. Thank you for the review, @mariahlaqua.

**Review by ghost — APPROVED** [review]

**Comment by RitaDee** [comment]
> Approved @RitaDee! Thanks again for taking this on, such a great improvement. I know @adiati98 is a bit busy this week, so it will be next week likely that we can progress! 🥇 😄

Noted. Thank you.

File: resources/frontend-development.md [hunk 2]
```diff
@@ -1,14 +1,70 @@
...
+The Odin Project provides a free and open-source curriculum for learning web development, covering HTML, CSS, JavaScript, and other essential technologies. Through project-based learning and guided tutorials, learners can build real-world projects and gain practical experience in frontend development, preparing them for careers in the tech industry.
+
 ## CSS
 
 Here is a list of useful websites for learning and mastering CSS:
 
--[Codecademy](https://www.codecademy.com/learn/learn-css) -[freeCodeCamp](https://www.freecodecamp.org/learn/) -[Flexbox Frogy](https://flexboxfroggy.com/) -[CSS Grid Garden](https://cssgridgarden.com/)
+- [Codecademy](https://www.codecademy.com/learn/learn-css) -[freeCodeCamp](https://www.freecodecamp.org/learn/)
+
+Codecademy offers an interactive platform for learning CSS, providing hands-on exercises and projects to reinforce concepts. From basic syntax to advanced styling techniques, this course guides learners through the essentials of Cascading Style Sheets, empowering them to create visually appealing web designs.
```
Comment (adiati98) [inline resources/frontend-development.md:43]: 1. There are two lists in one line here. Let's fix them.
2. Please give brief description for the freeCodeCamp CSS resource.
3. Please fix the format for all list with description in this page.
      
     See the format for the Codecademy here as example:
       
       1. Give a space after dash.
       2. Add double asterisks to bold the title.
       3. Add space after the title.
       4. Add three space by hitting the spacebar before writing the description.
       
    You can use [Markdown Live Preview](https://markdownlivepreview.com/) to see the format result of Markdown.

```suggestion
- **[Codecademy](https://www.codecademy.com/learn/learn-css)**
   
   Codecademy offers an interactive platform for learning CSS, providing hands-on exercises and projects to reinforce concepts. From basic syntax to advanced styling techniques, this course guides learners through the essentials of Cascading Style Sheets, empowering them to create visually appealing web designs.

- **[freeCodeCamp](https://www.freecodecamp.org/learn/)**
```

File: resources/frontend-development.md [hunk 3]
```diff
@@ -1,14 +1,70 @@
...
+
+- [Flexbox Frogy](https://flexboxfroggy.com/)
+
+Explore Flexbox Froggy, an interactive game that teaches Flexbox layout techniques through gamified challenges. By guiding frogs to their lily pads using Flexbox properties, users gain a hands-on understanding of Flexbox's powerful capabilities for building responsive web layouts.
```
Comment (adiati98) [inline resources/frontend-development.md:47]:
```suggestion
Explore Flexbox Froggy, an interactive game that teaches CSS flexbox layout techniques through gamified challenges. By guiding frogs to their lily pads using flexbox properties, users gain a hands-on understanding of flexbox's powerful capabilities for building responsive web layouts.
```

File: resources/frontend-development.md [hunk 4]
```diff
@@ -1,14 +1,70 @@
...
+
+-[CSS Grid Garden](https://cssgridgarden.com/)
+
+CSS Grid Garden offers an engaging way to learn CSS Grid Layout through a series of garden-themed challenges. By planting and arranging trees using CSS Grid properties, participants master the intricacies of grid-based layouts and gain confidence in designing complex web interfaces.
```
Comment (adiati98) [inline resources/frontend-development.md:51]:
```suggestion
CSS Grid Garden offers an engaging way to learn CSS grid through a series of garden-themed challenges. By planting and arranging trees using grid properties, participants master the intricacies of grid-based layouts and gain confidence in designing complex web interfaces.
```

File: resources/frontend-development.md [hunk 5]
```diff
@@ -1,14 +1,70 @@
...
+
+A list of useful resources for learning JavaScript:
+
+- [MDN Web Docs - JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
+
+MDN Web Docs offers an extensive resource on JavaScript, providing detailed documentation, tutorials, and references for developers. From basic syntax explanations to advanced topics like asynchronous programming, developers can rely on MDN for accurate and up-to-date information on JavaScript features and APIs.
+
+### JavaScript Library
+
+A javaScript library is a toolbox of pre-written code designed for a specific task. In other words,it's a collection of reusable functions and features framed or crafted to make certain programming tasks easier. For a beginner, think of it as a set of tools that you can use to build a web application without having to create everything from scratch. An example of a JavaScript library is React.
```
Comment (adiati98) [inline resources/frontend-development.md:65]:
```suggestion
A JavaScript library is a toolbox of pre-written code designed for a specific task. In other words, it's a collection of reusable functions and features framed or crafted to make certain programming tasks easier. For beginners, think of it as a set of tools that you can use to build a web application without having to create everything from scratch. An example of a JavaScript library is React.
```

File: resources/frontend-development.md [hunk 6]
```diff
@@ -1,14 +1,70 @@
...
+CSS Grid Garden offers an engaging way to learn CSS Grid Layout through a series of garden-themed challenges. By planting and arranging trees using CSS Grid properties, participants master the intricacies of grid-based layouts and gain confidence in designing complex web interfaces.
 
 ## JavaScript
 
-### React
+JavaScript is a scripting or programming language that allows you to create dynamic web pages.
+
+A list of useful resources for learning JavaScript:
+
+- [MDN Web Docs - JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
```
Comment (adiati98) [inline resources/frontend-development.md:59]:
```suggestion
- **[MDN Web Docs - JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)**
```

File: resources/frontend-development.md [hunk 7]
```diff
@@ -1,14 +1,70 @@
...
--[Codecademy](https://www.codecademy.com/learn/learn-css) -[freeCodeCamp](https://www.freecodecamp.org/learn/) -[Flexbox Frogy](https://flexboxfroggy.com/) -[CSS Grid Garden](https://cssgridgarden.com/)
+- [Codecademy](https://www.codecademy.com/learn/learn-css) -[freeCodeCamp](https://www.freecodecamp.org/learn/)
+
+Codecademy offers an interactive platform for learning CSS, providing hands-on exercises and projects to reinforce concepts. From basic syntax to advanced styling techniques, this course guides learners through the essentials of Cascading Style Sheets, empowering them to create visually appealing web designs.
+
+- [Flexbox Frogy](https://flexboxfroggy.com/)
+
+Explore Flexbox Froggy, an interactive game that teaches Flexbox layout techniques through gamified challenges. By guiding frogs to their lily pads using Flexbox properties, users gain a hands-on understanding of Flexbox's powerful capabilities for building responsive web layouts.
+
+-[CSS Grid Garden](https://cssgridgarden.com/)
```
Comment (adiati98) [inline resources/frontend-development.md:49]: 1. Add a space after dash.
2. Bold the title.

```suggestion
- **[CSS Grid Garden](https://cssgridgarden.com/)**
```

File: resources/frontend-development.md [hunk 8]
```diff
@@ -1,14 +1,70 @@
...
 ## CSS
 
 Here is a list of useful websites for learning and mastering CSS:
 
--[Codecademy](https://www.codecademy.com/learn/learn-css) -[freeCodeCamp](https://www.freecodecamp.org/learn/) -[Flexbox Frogy](https://flexboxfroggy.com/) -[CSS Grid Garden](https://cssgridgarden.com/)
+- [Codecademy](https://www.codecademy.com/learn/learn-css) -[freeCodeCamp](https://www.freecodecamp.org/learn/)
+
+Codecademy offers an interactive platform for learning CSS, providing hands-on exercises and projects to reinforce concepts. From basic syntax to advanced styling techniques, this course guides learners through the essentials of Cascading Style Sheets, empowering them to create visually appealing web designs.
+
+- [Flexbox Frogy](https://flexboxfroggy.com/)
```
Comment (adiati98) [inline resources/frontend-development.md:45]:
```suggestion
- **[Flexbox Frogy](https://flexboxfroggy.com/)**
```

File: resources/frontend-development.md [hunk 9]
```diff
@@ -1,14 +1,70 @@
...
+
+#### HTML (HyperText Markup Language)
+
+This is a markup language used to create the structure and layout of web pages.
+
+#### CSS (Cascading Style Sheets)
+
+It controls the presentation, formatting, and appearance of web pages, such as colors, fonts, and layout.
+
+#### JavaScript
```
Comment (adiati98) [inline resources/frontend-development.md:23]: There is a JavaScript section starts from line 53.

Same as the CSS, if there is no reason for duplication, let's move them here.

File: resources/frontend-development.md [hunk 10]
```diff
@@ -1,14 +1,70 @@
...
+
+A list of useful websites to gain mastery of frontend development:
+
+- [Frontend Masters](https://frontendmasters.com/):
+
+Frontend Masters is a comprehensive learning platform for frontend developers, offering in-depth courses on a wide range of topics, including HTML, CSS, JavaScript, and modern frontend frameworks. With expert-led instruction and hands-on projects, developers can enhance their skills and stay up-to-date with the latest trends in frontend development.
+
+- [The Odin Project](https://www.theodinproject.com/):
+
+The Odin Project provides a free and open-source curriculum for learning web development, covering HTML, CSS, JavaScript, and other essential technologies. Through project-based learning and guided tutorials, learners can build real-world projects and gain practical experience in frontend development, preparing them for careers in the tech industry.
```
Comment (adiati98) [inline resources/frontend-development.md:35]: Let's move these resources to right after the title of "Resources to Learn Frontend Development".

File: resources/frontend-development.md [hunk 11]
```diff
@@ -1,14 +1,70 @@
...
+
+#### JavaScript
+
+Enables dynamic interactions and functionality on web pages, such as form validation, animations, and data manipulation. JavaScript frameworks and libraries like React, Vue.js, and Angular facilitate the development of complex user interfaces and enhance interactivity.
+
+A list of useful websites to gain mastery of frontend development:
+
+- [Frontend Masters](https://frontendmasters.com/):
+
+Frontend Masters is a comprehensive learning platform for frontend developers, offering in-depth courses on a wide range of topics, including HTML, CSS, JavaScript, and modern frontend frameworks. With expert-led instruction and hands-on projects, developers can enhance their skills and stay up-to-date with the latest trends in frontend development.
```
Comment (adiati98) [inline resources/frontend-development.md:31]:
```suggestion
- **[Frontend Masters](https://frontendmasters.com/)**

Frontend Masters is a comprehensive learning platform for frontend developers, offering in-depth courses on a wide range of topics, including HTML, CSS, JavaScript, and modern frontend frameworks. With expert-led instruction and hands-on projects, developers can enhance their skills and stay up-to-date with the latest trends in frontend development.
```

File: resources/frontend-development.md (hunk 10 above)
Comment (adiati98) [inline resources/frontend-development.md:35]:
```suggestion
- **[The Odin Project](https://www.theodinproject.com/)**

The Odin Project is an open source that provides free curriculum for learning web development, covering HTML, CSS, JavaScript, and other essential technologies. Through project-based learning and guided tutorials, learners can build real-world projects and gain practical experience in frontend development, preparing them for careers in the tech industry.
```

**Review by adiati98 — CHANGES_REQUESTED** [review]
Hey @RitaDee,

Thank you for your PR and patience! 😄

I left reviews here. Please read them through and fix them. If there's anything that needs clarification, feel free to ask and tag me! 😊

Other than things in the review, please update your PR form to inlcude these:

- Change "Relates to" to "Closes" before the issue link.
- I see that you made changes beyond what is asked in the issue. But we agreed that these are great addition to the knowledge-hub! ✨
   To reflect these additions, please update the PR description to explain your thoughts behind them (like why did you add them, the  benefits, anything!). That way, we can have the history of why we implement these changes,

Thank you!

**Comment by RitaDee** [comment]
> Hey @RitaDee,
> 
> Thank you for your PR and patience! 😄
> 
> I left reviews here. Please read them through and fix them. If there's anything that needs clarification, feel free to ask and tag me! 😊
> 
> Other than things in the review, please update your PR form to inlcude these:
> 
> * Change "Relates to" to "Closes" before the issue link.
> * I see that you made changes beyond what is asked in the issue. But we agreed that these are great addition to the knowledge-hub! ✨
>   To reflect these additions, please update the PR description to explain your thoughts behind them (like why did you add them, the  benefits, anything!). That way, we can have the history of why we implement these changes,
> 
> Thank you!

Thank you, @adiati98, for the review. I will carefully implement the changes.

File: resources/frontend-development.md [hunk 12]
```diff
@@ -1,14 +1,70 @@
...
 
+Frontend Development is a pivotal aspect of web development that focuses on crafting the visual and interactive components of a web application. It encompasses two fundamental elements: the graphical design and the user interface (UI) development.
+
+## Graphical Design
+
+In the context of frontend development, graphical design involves conceptualizing and creating the visually appealing layouts, typography, color schemes, and graphical elements that collectively form the aesthetic foundation of the application. Designers employ tools like Adobe Photoshop, Sketch, or Figma to translate ideas into tangible visual assets.
+
+## User Interface (UI) Development
+
+UI development focuses on implementing the graphical design into functional, interactive interfaces that users can seamlessly navigate and engage with. It includes writing codes using technologies like HTML, CSS, and JavaScript to structure content, style elements, and add dynamic behavior.
```
[the remark this answers is not in the document]
Comment (RitaDee) [reply resources/frontend-development.md:11]: Done.

File: resources/frontend-development.md (hunk 2 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done.

File: resources/frontend-development.md [hunk 13]
```diff
@@ -1,14 +1,70 @@
...
+
+### Key Aspects of Frontend Development
+
+#### HTML (HyperText Markup Language)
+
+This is a markup language used to create the structure and layout of web pages.
+
+#### CSS (Cascading Style Sheets)
+
+It controls the presentation, formatting, and appearance of web pages, such as colors, fonts, and layout.
```
[the remark this answers is not in the document]
Comment (RitaDee) [reply resources/frontend-development.md:21]: Done.

File: resources/frontend-development.md (hunk 3 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done.

File: resources/frontend-development.md (hunk 4 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done.

File: resources/frontend-development.md [hunk 14]
```diff
@@ -1,14 +1,70 @@
...
+
+## Graphical Design
+
+In the context of frontend development, graphical design involves conceptualizing and creating the visually appealing layouts, typography, color schemes, and graphical elements that collectively form the aesthetic foundation of the application. Designers employ tools like Adobe Photoshop, Sketch, or Figma to translate ideas into tangible visual assets.
+
+## User Interface (UI) Development
+
+UI development focuses on implementing the graphical design into functional, interactive interfaces that users can seamlessly navigate and engage with. It includes writing codes using technologies like HTML, CSS, and JavaScript to structure content, style elements, and add dynamic behavior.
+
+### Key Aspects of Frontend Development
```
[the remark this answers is not in the document]
Comment (RitaDee) [reply resources/frontend-development.md:13]: Done.

File: resources/frontend-development.md (hunk 5 above)
Comment (RitaDee) [reply resources/frontend-development.md:67]: Done.

File: resources/frontend-development.md (hunk 6 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done.

File: resources/frontend-development.md (hunk 9 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done.

File: resources/frontend-development.md (hunk 10 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done.

Comment (RitaDee) [reply resources/frontend-development.md]: Done.

File: resources/frontend-development.md (hunk 11 above)
Comment (RitaDee) [reply resources/frontend-development.md]: Done.

**Comment by mariahlaqua** [comment]
Lovely PR description @RitaDee, thanks for giving us insight into your motivation and thought process. Super happy that you contributed this. 😄

**Comment by RitaDee** [comment]
> Lovely PR description @RitaDee, thanks for giving us insight into your motivation and thought process. Super happy that you contributed this. 😄

Thank you, @mariahlaqua. I am glad to be a part of this super team.

**Review by adiati98 — APPROVED** [review]
Hey @RitaDee,
Thank you for the changes and for updating the PR description! 🙌
I really appreciate the thoughts behind it. ✨

LGTM! 🚀

go time GIF

**Comment by RitaDee** [comment]
> Hey @RitaDee, Thank you for the changes and for updating the PR description! 🙌 I really appreciate the thoughts behind it. ✨
> 
> LGTM! 🚀
> 
> go time GIF

Thank you, Ayu.

**Pull request merged**